### PR TITLE
Adds Jump Boots

### PR DESCRIPTION
**Pull request by MMMiracles**
:cl: MMMiracles
add: Adds jump boots, a mid/end-tier mining equipment that allows the user to dash over 4-wide chasms.
/:cl:

Jump boots are a 2500 point item in the mining vendor that allows the user to dash over 4-wide chasms without worry of falling in mid-way. There is a 6-second delay between each dash to prevent rapid (wave)dashing down halls or something equally silly.

An upgrade can be purchased as well from the vendor that decreases the recharge rate by a second and allows dashing over lava unharmed. Probably unbalanced but whatever it's FUN.

Trying to dash into walls is ill-advised.

jetboot
Action Icon
jetbootitem
Item Icon

### CODE REVIEW

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 1]
```diff
@@ -161,3 +161,43 @@
...
+	burn_state = FIRE_PROOF
+	pockets = /obj/item/weapon/storage/internal/pocket/shoes
+	actions_types = list(/datum/action/item_action/bhop)
+	var/recharging = FALSE
+	var/jumping = FALSE
+
+/obj/item/clothing/shoes/bhop/ui_action_click(mob/user, actiontype)
+	bhop()
+
+/obj/item/clothing/shoes/bhop/verb/bhop()
```
Comment (RemieRichards) [inline code/modules/clothing/shoes/miscellaneous.dm]: no.

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 2]
```diff
@@ -161,3 +161,43 @@
...
+
+	var/atom/target = get_edge_target_turf(usr, usr.dir)
+
+	jumping = TRUE
+	playsound(src.loc, 'sound/effects/stealthoff.ogg', 50, 1, 1)
+	usr.visible_message("<span class='warning'>[usr] dashes foward into the air!</span>")
+	usr.throw_at(target,3,1, spin=0, diagonals_first = 1)
+	recharging = TRUE
+	jumping = FALSE
+	spawn(60)
```
Comment (RemieRichards) [inline code/modules/clothing/shoes/miscellaneous.dm]: timer subsystem.

File: code/modules/mining/mine_turfs.dm [hunk 3]
```diff
@@ -605,6 +605,10 @@
 			return 0
 	if(ishuman(AM))
 		var/mob/living/carbon/human/H = AM
+		if(istype(H.shoes, /obj/item/clothing/shoes/bhop))
```
Comment (RemieRichards) [inline code/modules/mining/mine_turfs.dm]: shouldn't be necessary, the boots throw you over the chasm, and Joan recently made thrown things ignore chasm falling.

Comment (MMMiracles) [inline code/modules/mining/mine_turfs.dm]: did not know that, will fix.

File: code/modules/clothing/shoes/miscellaneous.dm (hunk 1 above)
Comment (MMMiracles) [inline code/modules/clothing/shoes/miscellaneous.dm]: Is the verb not required and can I just do that inside the ui_action_click or is there just a different way of doing so?

Comment (RemieRichards) [inline code/modules/clothing/shoes/miscellaneous.dm]: verb is 100% not needed.

put all the code (besides verb-specific stuff) in ```ui_action_click()```, and swap ```usr``` for ```owner```

**Comment by Man-Shroom-** [comment]
doesn't work on 1 or 2 tile wide lava spots?

**Comment by jughu** [comment]
Cant we just add this to rnd why does miners get all the cool shit

**Comment by MMMiracles** [comment]
@Man-Shroom- lava seems like its suppose to be more of a hard counter past a few things, might add it but meh.

@jughu why would RnD need the ability to mass-produce an item that currently only has any actual use in lavaland, and even then it's limited.

**Comment by optimumtact** [comment]
@jughu RnD has had most of the cool shit for about 3+ years now, it's about time other jobs got love.

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 4]
```diff
@@ -161,3 +161,33 @@
...
+	desc = "A specialized pair of combat boots with a built-in propulsion system for rapid foward movement."
+	icon_state = "jetboots"
+	burn_state = FIRE_PROOF
+	pockets = /obj/item/weapon/storage/internal/pocket/shoes
+	actions_types = list(/datum/action/item_action/bhop)
+	var/recharging_time = 0
+	var/jumping = FALSE
+
+/obj/item/clothing/shoes/bhop/ui_action_click(mob/user, actiontype)
+	if(!istype(usr, /mob/living))
```
Comment (Fox-McCloud) [inline code/modules/clothing/shoes/miscellaneous.dm]: `!isliving(usr)`

**Comment by hornygranny** [comment]
You kiss the feet of Hermes! A pair of winged sandals appear next to the feet of Hermes.
hqdefault

**Comment by KorPhaeron** [comment]
Dash should probably be farther, 2 wide chasm is rare

**Comment by AdamElTablawy** [comment]
Only if the autodrobe gets blue overalls and a red plumber hat

**Comment by Zilenan91** [comment]
The changelog still says 2-wide

**Comment by Shadowlight213** [comment]
I was gonna make a lavaland variant of wings for going over lava but I was waiting for @KorPhaeron to stop rigging his patreon art raffles so shaps wins in order to get the sprites.

**Comment by MMMiracles** [comment]
jump boots > wings

**Comment by Yiu113** [comment]
Jump boots are better than wings because I can be a dirty ligger and still cross chasms.

File: code/game/turfs/simulated/floor/plating.dm [hunk 5]
```diff
@@ -269,6 +269,12 @@
 					var/mob/living/live = L.buckled
 					if("lava" in live.weather_immunities)
 						continue
+			if(ishuman(L))
```
Comment (KorPhaeron) [inline code/game/turfs/simulated/floor/plating.dm]: this is pretty snowflake

Comment (MMMiracles) [inline code/game/turfs/simulated/floor/plating.dm]: chasms do the same thing for jaunters, if there's a less snowflake-y method then sure.

Comment (RemieRichards) [inline code/game/turfs/simulated/floor/plating.dm]: just give thrown things protection from lava, thrown OBJECTS already are, but lava still burns thrown mobs, fix that and you avoid snowflake.

Comment (KorPhaeron) [inline code/game/turfs/simulated/floor/plating.dm]: that'd require a remap of the gulag

Comment (KorPhaeron) [inline code/game/turfs/simulated/floor/plating.dm]: I'll merge this tonight if you want to remove the lava upgrade and figure it out in another PR

File: code/modules/mining/machine_vending.dm [hunk 6]
```diff
@@ -45,8 +45,7 @@
 		new /datum/data/mining_equipment("Drone Health Upgrade",/obj/item/device/mine_bot_ugprade/health,      			   	       		400),
 		new /datum/data/mining_equipment("Drone Ranged Upgrade",/obj/item/device/mine_bot_ugprade/cooldown,      			   	   		600),
 		new /datum/data/mining_equipment("Drone AI Upgrade",    /obj/item/slimepotion/sentience/mining,      			   	      		1000),
-		new /datum/data/mining_equipment("Jump Boots",    		/obj/item/clothing/shoes/bhop,      			   	      				2500),
-		new /datum/data/mining_equipment("Jump Boots Lava Upgrade", /obj/item/jumpbootupgrade,      									1000),
+		new /datum/data/mining_equipment("Jump Boots",    		/obj/item/clothing/shoes/bhop,      			   	      				2500),      									1000),
```
Comment (KorPhaeron) [inline code/modules/mining/machine_vending.dm]: you left 1000) here

**Pull request merged**